src/fw-logs/fw-log-data.h

```diff
@@ -17,89 +17,61 @@ namespace librealsense
             uint32_t get_timestamp() const;
         };
 
-        rs2_log_severity fw_logs_severity_to_log_severity(int32_t severity);
+        rs2_log_severity fw_logs_severity_to_rs2_log_severity( int32_t severity );
+        rs2_log_severity legacy_fw_logs_severity_to_rs2_log_severity( int32_t severity );
 
-        static const int BINARY_DATA_SIZE = 20;
-
-        typedef union
-        {
-            uint32_t value;
-            struct
-            {
-                uint32_t magic_number : 8;
-                uint32_t severity : 5;
-                uint32_t thread_id : 3;
-                uint32_t file_id : 11;
-                uint32_t group_id : 5;
-            } bits;
-        } fw_log_header_dword1;
-
-        typedef union
+        struct fw_log_binary
         {
-            uint32_t value;
-            struct
-            {
-                uint32_t event_id : 16;
-                uint32_t line_id : 12;
-                uint32_t seq_id : 4;
-            } bits;
-        } fw_log_header_dword2;
+            uint32_t magic_number : 8;
+            uint32_t severity : 5;
+            uint32_t source_id : 3;  // SoC ID in D500, thread ID in D400
+            uint32_t file_id : 11;
+            uint32_t module_id : 5;
+            uint32_t event_id : 16;
+            uint32_t line_id : 12;
+            uint32_t seq_id : 4;  // Rolling counter 0-15 per module
+        };
 
-        struct fw_log_header_dword3
+        struct legacy_fw_log_binary : public fw_log_binary
         {
             uint16_t p1;
             uint16_t p2;
-        };
-
-        struct fw_log_header_dword4
-        {
             uint32_t p3;
-        };
-
-        struct fw_log_header_dword5
-        {
             uint32_t timestamp;
         };
 
-        struct fw_log_binary
+        struct extended_fw_log_binary : public fw_log_binary
```

Comment:
I tried to avoid model names as it might not be correct.
The new (extended) format aims to be future proof (within reason). What if this format would be used for D600 models? What about old L500 models using the old format.
But I agree that "legacy" and "extended" are also not that good, as further changes to the format might make these names obsolete.

Lets talk tomorrow and decide on the preferred naming approach.

Reply:
After discussion the naming convention is - "legacy" parser/device/etc.. will have no prefix. New format classes will use "extended_" prefix.
This cause some weird naming in the implementation as I had `firmware_logger_device` inheriting `extended_firmware_logger_device`. I have changed the implementation moving functions from base classes to inheritors to reverse this. Now "extended_" classes inherit the no prefix classes.